### ISSUE TITLE
[false-positive] “isset(...) usage” incorrectly recommending `null === ...` 

### ISSUE THREAD

**Issue by brandonkelly** [kalessil/phpinspectionsea#105]
Example:

```php
class Foo
{
    public function bar()
    {
        if (random_int(1, 100) % 2) {
            $maybe = true;
        }

        if (!isset($maybe)) {
            $maybe = false;
        }

        return $maybe;
    }
}
```

Silly example but this illustrates the bug well: `$maybe` may or may not have been defined so `isset()` is needed. If that line was changed to `if (null === $maybe)` you would get a PHP notice about `$maybe` being undefined when the first condition doesn’t pass.

**Comment by funivan** [comment]
Hello. In your case `$maybe` is always initialized in the return statement =) 
Previous if statement said: If our `$maybe` variable is not initialized then init it with value `false`
So all is ok. 
And I can guess that this inspection is standard phpstorm inspection: `Undefined variable`

**Comment by kalessil** [comment]
This would be a good refactoring candidate: `$maybe = (1 === random_int(1, 100) % 2)`.

Apart of the idea, I should say that the inspection has no any possibility to find out if variable is defined or not when reaching the isset construction. Pity, but that's it's API constraints.

**Comment by mlamm** [comment]
Hey, i just ran into this issue, I see there's no way for you to detect if that variable is defined or not, however, are there other options to make that issue less of an issue, like changing the wording of the warning?
